runners/core-java/src/main/java/org/apache/beam/runners/core/ReduceFnRunner.java

```diff
@@ -957,6 +958,7 @@ private Instant onTrigger(
     // Extract the window hold, and as a side effect clear it.
     final WatermarkHold.OldAndNewHolds pair =
         watermarkHold.extractAndRelease(renamedContext, isFinished).read();
+    // TODO: This isn't accurate if the elements are late. See BEAM-2262
```

Comment:
+1 to this - now that GC is done via window expiration rather than per-element lateness there is no need to relocate output times to EOW even when we relocate the hold to GC time